### PR TITLE
Bug fixes & more verbose

### PR DESCRIPTION
**Pull request by g0tmi1k**
...sorry for merging it all into one commit

### CODE REVIEW

**Comment by MysticRyuujin** [comment]
Does anything in the readme need to get updated for clarity on the fixes for Debian? Otherwise I love it.

**Comment by g0tmi1k** [comment]
Just done a quick once over and It shouldn't do (not until the next version of guacamole is out)!

File: guac-install.sh [hunk 1]
```diff
@@ -499,8 +501,15 @@ fi
...
-if [ "$installMySQL" = true ]; then
-    # restart mysql
+if [ "${installMySQL}" = true ]; then
+
+    if [ -e /etc/mysql/mariadb.conf.d/50-server.cnf ]; then
+    elif [ -e /etc/mysql/mariadb.conf.d/50-server.cnf ]; then
+    else
+    fi
+
+
```
Comment (MysticRyuujin) [inline guac-install.sh]: What happened here?

File: guac-install.sh [hunk 2]
```diff
@@ -225,44 +233,59 @@ elif [[ "${NAME}" == *"Debian"* ]] || [[ "${NAME}" == *"Raspbian GNU/Linux"* ]]
...
 # fallback is tomcat7
-if [[ $(apt-cache show tomcat9 2> /dev/null | egrep "Version: 9" | wc -l) -gt 0 ]]; then
+if [[ $( apt-cache show tomcat9 2> /dev/null | egrep "Version: 9" | wc -l ) -gt 0 ]]; then
+    echo -e "${BLUE}Found tomcat9 package...${NC}"
     TOMCAT="tomcat9"
-elif [[ $(apt-cache show tomcat8 2> /dev/null | egrep "Version: 8.[5-9]" | wc -l) -gt 0 ]]; then
+elif [[ $( apt-cache show tomcat8 2> /dev/null | egrep "Version: 8.[5-9]" | wc -l ) -gt 0 ]]; then
+    echo -e "${BLUE}Found tomcat8 package...${NC}"
+    TOMCAT="tomcat8"
+elif [[ $( apt-cache show tomcat7 2> /dev/null | egrep "Version: 8.[5-9]" | wc -l ) -gt 0 ]]; then
```
Comment (MysticRyuujin) [inline guac-install.sh]: This is still looking for Tomcat7 but grepping for version 8?

**Comment by MysticRyuujin** [comment]
More comments added

**Comment by g0tmi1k** [comment]
Sorry, I don't follow

**Comment by MysticRyuujin** [comment]
Scroll up? I added comments inline in review

**Comment by g0tmi1k** [comment]
I don't see anything (nor have I had any emails).
Able to direct link?

**Comment by MysticRyuujin** [comment]
Sorry forgot to hit submit

File: guac-install.sh [hunk 3]
```diff
@@ -1,13 +1,17 @@
 #!/bin/bash
 
 # Check if user is root or sudo
-if ! [ $(id -u) = 0 ]; then echo "Please run this script as sudo or root"; exit 1 ; fi
+if ! [ $(id -u) = 0 ]; then echo "Please run this script as sudo or root" 1>&2 ; exit 1 ; fi
+
+# Check to see if any old files left over
+if [ "$( find . -maxdepth 1 \( -name 'guacamole-*' -o -name 'mysql-connector-java-*' \) )" != "" ]; then echo "Possible temp files detected. Please review 'guacamole-*' & 'mysql-connector-java-*'" 1>&2 ; exit 1 ; fi
```
[the remark this answers is not in the document]
Comment (g0tmi1k) [reply guac-install.sh]: Sure!

File: guac-install.sh [hunk 4]
```diff
@@ -234,35 +242,50 @@ elif [[ "${NAME}" == *"Debian"* ]] || [[ "${NAME}" == *"Raspbian GNU/Linux"* ]]
...
 # tomcat9 is the latest version
 # tomcat8.0 is end of life, but tomcat8.5 is current
 # fallback is tomcat7
 if [[ $(apt-cache show tomcat9 2> /dev/null | egrep "Version: 9" | wc -l) -gt 0 ]]; then
+    echo -e "${BLUE}Found tomcat9 package...${NC}"
     TOMCAT="tomcat9"
 elif [[ $(apt-cache show tomcat8 2> /dev/null | egrep "Version: 8.[5-9]" | wc -l) -gt 0 ]]; then
+    echo -e "${BLUE}Found tomcat8 package...${NC}"
+    TOMCAT="tomcat8"
+elif [[ $(apt-cache show tomcat7 2> /dev/null | egrep "Version: 8.[5-9]" | wc -l) -gt 0 ]]; then
```
[the remark this answers is not in the document]
Comment (g0tmi1k) [reply guac-install.sh]: Ahhh! Yes. Good catch.
Will address
_bad copy/paste_

File: guac-install.sh [hunk 5]
```diff
@@ -234,35 +242,50 @@ elif [[ "${NAME}" == *"Debian"* ]] || [[ "${NAME}" == *"Raspbian GNU/Linux"* ]]
...
-# Update apt so we can search apt-cache for newest tomcat version supported & libmysql-java
+# Update apt so we can search apt-cache for newest Tomcat version supported & libmariadb-java/libmysql-java
 echo -e "${BLUE}Updating apt...${NC}"
 apt-get -qq update
 
-# Check if libmysql-java is available
-if [[ $(apt-cache show libmysql-java 2> /dev/null | egrep "Version:" | wc -l) -gt 0 ]]; then
+# Check if libmariadb-java/libmysql-java is available
+# Debian 10 >= ~ https://packages.debian.org/search?keywords=libmariadb-java
+if [[ $(apt-cache show libmariadb-java 2> /dev/null | egrep "Version:" | wc -l) -gt 0 ]]; then
```
[the remark this answers is not in the document]
Comment (g0tmi1k) [reply guac-install.sh]: I just re-used the same code from before - which is working for the detection on Debian 10 (not sure if its required or not tho!).

Will investigate

File: guac-install.sh [hunk 6]
```diff
@@ -234,35 +242,50 @@ elif [[ "${NAME}" == *"Debian"* ]] || [[ "${NAME}" == *"Raspbian GNU/Linux"* ]]
...
+    # When something higer than 1.1.0 is out ~ https://issues.apache.org/jira/browse/GUACAMOLE-852
+    #echo -e "${BLUE}Found libmariadb-java package...${NC}"
+    #LIBJAVA="libmariadb-java"
+    # For v1.1.0 and lower
+    echo -e "${YELLOW}Found libmariadb-java package (known issues). Will download libmysql-java ${MCJVER} and install manually${NC}"
+    LIBJAVA=""
+# Debian 9 <=  ~ https://packages.debian.org/search?keywords=libmysql-java
+elif [[ $(apt-cache show libmysql-java 2> /dev/null | egrep "Version:" | wc -l) -gt 0 ]]; then
```
[the remark this answers is not in the document]
Comment (g0tmi1k) [reply guac-install.sh]: Same comment as before ~ https://github.com/MysticRyuujin/guac-install/pull/123#discussion_r394312645

File: guac-install.sh (hunk 1 above)
Comment (g0tmi1k) [reply guac-install.sh]: Doh!
This wasn't  meant to be in this commit (miss staged it).

This was meant to be in the next commit to address ~ 
 https://issues.apache.org/jira/browse/GUACAMOLE-760

File: guac-install.sh (hunk 2 above)
Comment (g0tmi1k) [reply guac-install.sh]: Correct - re-used the code without altering.

Will address

Good catch

File: guac-install.sh [hunk 7]
```diff
@@ -225,44 +239,59 @@ elif [[ "${NAME}" == *"Debian"* ]] || [[ "${NAME}" == *"Raspbian GNU/Linux"* ]]
...
-if [[ $(apt-cache show tomcat9 2> /dev/null | egrep "Version: 9" | wc -l) -gt 0 ]]; then
+if [[ $( apt-cache show tomcat9 2> /dev/null | egrep "Version: 9" | wc -l ) -gt 0 ]]; then
+    echo -e "${BLUE}Found tomcat9 package...${NC}"
     TOMCAT="tomcat9"
-elif [[ $(apt-cache show tomcat8 2> /dev/null | egrep "Version: 8.[5-9]" | wc -l) -gt 0 ]]; then
+elif [[ $( apt-cache show tomcat8 2> /dev/null | egrep "Version: 8.[5-9]" | wc -l ) -gt 0 ]]; then
+    echo -e "${BLUE}Found tomcat8.5+ package...${NC}"
+    TOMCAT="tomcat8"
+elif [[ $( apt-cache show tomcat7 2> /dev/null | egrep "Version: 7" | wc -l ) -gt 0 ]]; then
+    echo -e "${BLUE}Found tomcat7 package...${NC}"
```
Comment (MysticRyuujin) [inline guac-install.sh]: tomcat7 found but setting TOMCAT="tomcat8" next line

File: guac-install.sh [hunk 8]
```diff
@@ -568,6 +642,9 @@ echo
 # Done
 echo -e "${BLUE}Installation Complete\n- Visit: http://localhost:8080/guacamole/\n- Default login (username/password): guacadmin/guacadmin\n***Be sure to change the password***.${NC}"
 
-if [ "$installDuo" = true ]; then
+if [ "${installDuo}" = true ]; then
     echo -e "${YELLOW}\nDon't forget to configure Duo in guacamole.properties. You will not be able to login otherwise.\nhttps://guacamole.apache.org/doc/${GUACVERSION}/gug/duo-auth.html${NC}"
 fi
+
+sleep 2s
+xdg-open http://localhost:8080/guacamole/
```
Comment (MysticRyuujin) [inline guac-install.sh]: Not sure what this is? Is it trying to open the URL from the system? I imagine most people run this headless.

Comment (g0tmi1k) [reply guac-install.sh]: Correct
And most people also don't use root as default - so it wouldn't work

File: guac-install.sh (hunk 7 above)
Comment (g0tmi1k) [reply guac-install.sh]: urgh!
Good eye - fixed the typo

**Comment by g0tmi1k** [comment]
Commits done :)
I believe my review is done too.

Sorry for spamming over multiple commits.

**Comment by MysticRyuujin** [comment]
All good. Very much appreciated.

**Comment by g0tmi1k** [comment]
🎉 Thank you!

**Pull request merged**